### PR TITLE
Feature/issue 1657

### PR DESCRIPTION
**Pull request by shshshl09**

### CODE REVIEW

File: nugu-agent/src/main/java/com/skt/nugu/sdk/agent/permission/RequestPermissionDirectiveHandler.kt [hunk 1]
```diff
@@ -48,7 +49,8 @@ class RequestPermissionDirectiveHandler(
     }
 
     override fun handleDirective(info: DirectiveInfo) {
-        val payload = MessageFactory.create(info.directive.payload, Payload::class.java)
+        val payload: Payload? = parsePayload(info.directive.payload)
+
```
Comment (tommy5oo) [inline nugu-agent/src/main/java/com/skt/nugu/sdk/agent/permission/RequestPermissionDirectiveHandler.kt:53]: Why parse the payload manually?

Comment (shshshl09) [reply nugu-agent/src/main/java/com/skt/nugu/sdk/agent/permission/RequestPermissionDirectiveHandler.kt:53]: At gson(default setting), set to null if field does not exist or invalid value, even if nonnull field.
It is more complicate to implement custom gson serializer/deserializer.

Comment (shshshl09) [reply nugu-agent/src/main/java/com/skt/nugu/sdk/agent/permission/RequestPermissionDirectiveHandler.kt:53]: Before changes, RequestPermissionDirectiveHandlerTest.testHandleDirectiveOnFailed() failed. result.setCompleted() called.

Comment (tommy5oo) [reply nugu-agent/src/main/java/com/skt/nugu/sdk/agent/permission/RequestPermissionDirectiveHandler.kt:53]: OK, thanks for the explanation.

**Review by tommy5oo — APPROVED** [review]

**Pull request merged**